src/bandit-evaluator.ts

```diff
@@ -173,6 +184,43 @@ export class BanditEvaluator {
     subjectKey: string,
     actionWeights: Record<string, number>,
   ): string {
-    return Object.keys(actionWeights)[0]; // TODO: math and stuff
+    // Deterministically "shuffle" the actions
+    // This way as action weights shift, a bunch of users who were on the edge of one action won't all be shifted to the
+    // same new action at the same time.
+    const shuffledActions = Object.entries(actionWeights).sort((a, b) => {
+      const actionAShard = this.sharder.getShard(
+        `${flagKey}-${subjectKey}-${a[0]}`,
+        this.assignmentShards,
+      );
+      const actionBShard = this.sharder.getShard(
+        `${flagKey}-${subjectKey}-${b[0]}`,
+        this.assignmentShards,
+      );
+      let result = actionAShard - actionBShard;
+      if (result === 0) {
+        // In the unlikely case of a tie in randomized assigned shards, break the tie with the action names
+        result = a[0] < b[0] ? -1 : 1;
+      }
+      return result;
```

Comment:
nit: could simplify with just returning with a ternary operator
```typescript
return actionAShard - actionBShard === 0 ? a[0] < b[0] ? -1 : 1 : result;

Reply:
True, but I like how the if statement let's me locate a comment about the tie-breaking in a sensible location